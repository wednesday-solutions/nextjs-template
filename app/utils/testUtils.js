import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import configureStore from '../configureStore';
import colors from '@themes/colors';
import { translationMessages, DEFAULT_LOCALE } from '../i18n';

export const getComponentStyles = (Component, props = {}) => {
  renderProvider(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);

  return window.getComputedStyle(componentRoots[0])._values;
};

export const renderProvider = (children) => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} key={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <ThemeProvider theme={colors}>{children}</ThemeProvider>
    </IntlProvider>
  );
};
export const timeout = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export const renderStoreProvider = (children) => {
  const store = configureStore({});
  return render(
    <Provider store={store}>
      <IntlProvider locale={DEFAULT_LOCALE} key={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
        <ThemeProvider theme={colors}>{children}</ThemeProvider>
      </IntlProvider>
    </Provider>
  );
};
