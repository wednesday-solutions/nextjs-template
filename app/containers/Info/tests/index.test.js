/**
 *
 * Tests for Info container
 *
 *
 */

import React from 'react'
// import { fireEvent } from '@testing-library/dom';
import { renderProvider } from '@utils/testUtils'
import { InfoTest as Info } from '../index'

describe('<Info /> container tests', () => {
  // let submitSpy

  beforeEach(() => {
    // submitSpy = jest.fn();
  })

  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(<Info />)
    expect(baseElement).toMatchSnapshot()
  })
})
