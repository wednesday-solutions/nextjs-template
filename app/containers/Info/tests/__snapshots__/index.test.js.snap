// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Info /> container tests should render and match the snapshot 1`] = `
.emotion-2.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: px;
  width: 100%;
  margin: 0 auto;
  padding: 20px;
}

<body>
  <div>
    <div
      class="ant-row ant-row-center ant-row-middle emotion-0"
      flex="1 1 90%"
      style="height: 100vh;"
    >
      <div
        class="ant-col emotion-0"
      >
        <div
          class="emotion-2"
          style="min-width: 30rem;"
        >
          <div
            class="ant-card ant-card-bordered emotion-0"
          >
            <div
              class="ant-card-head"
            >
              <div
                class="ant-card-head-wrapper"
              >
                <div
                  class="ant-card-head-title"
                >
                  <div
                    class="ant-row ant-row-space-between ant-row-middle emotion-0"
                    data-testid="title"
                  >
                    <div
                      class="ant-skeleton ant-skeleton-active emotion-0"
                    >
                      <div
                        class="ant-skeleton-content"
                      >
                        <h3
                          class="ant-skeleton-title"
                          style="width: 38%;"
                        />
                        <ul
                          class="ant-skeleton-paragraph"
                        >
                          <li />
                          <li />
                          <li
                            style="width: 61%;"
                          />
                        </ul>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-card-body"
            >
              <div
                class="ant-skeleton ant-skeleton-active emotion-0"
              >
                <div
                  class="ant-skeleton-content"
                >
                  <h3
                    class="ant-skeleton-title"
                    style="width: 38%;"
                  />
                  <ul
                    class="ant-skeleton-paragraph"
                  >
                    <li />
                    <li />
                    <li
                      style="width: 61%;"
                    />
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
